
/***************************************************************************
 *  MachineCommands.proto - LLSF Protocol - Machine Commands
 *
 *  Created: Thu Mar 07 16:00:55 2013
 *  Copyright  2013  Tim Niemueller [www.niemueller.de]
 ****************************************************************************/

/*  Redistribution and use in source and binary forms, with or without
 *  modification, are permitted provided that the following conditions
 *  are met:
 *
 * - Redistributions of source code must retain the above copyright
 *   notice, this list of conditions and the following disclaimer.
 * - Redistributions in binary form must reproduce the above copyright
 *   notice, this list of conditions and the following disclaimer in
 *   the documentation and/or other materials provided with the
 *   distribution.
 * - Neither the name of the authors nor the names of its contributors
 *   may be used to endorse or promote products derived from this
 *   software without specific prior written permission.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
 * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
 * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
 * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
 * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
 * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
 * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
 * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
 * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
 * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
 * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
 * OF THE POSSIBILITY OF SUCH DAMAGE.
 */

syntax = "proto2";

package llsf_msgs;

import "MachineInfo.proto";

option java_package = "org.robocup_logistics.llsf_msgs";
option java_outer_classname = "MachineCommandsProtos";

enum MachineState {
  IDLE      = 1;
  AVAILABLE = 2;
  PROCESSED = 3;
  DELIVERED = 4;
  RETRIEVED = 5;
  RESET     = 15;
}

// Load puck into machine area.
message SetMachineState {
  enum CompType {
    COMP_ID  = 2000;
    MSG_TYPE = 17;
  }

  // Machine to place puck at
  required string machine_name = 1;
  // State to set
  required MachineState state = 2;
}


// Load puck into machine area.
message MachineAddBase {
  enum CompType {
    COMP_ID  = 2000;
    MSG_TYPE = 18;
  }

  // Machine to place puck at
  required string machine_name = 1;
}

// Set machine lights.
message SetMachineLights {
  enum CompType {
    COMP_ID  = 2000;
    MSG_TYPE = 19;
  }

  // Machine to place puck at
  required string machine_name = 1;
  // Desired state of the lights
  repeated LightSpec lights = 2;
}
